proto/buf/alpha/registry/v1alpha1/convert.proto

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "buf/alpha/convert/v1alpha1/convert.proto";
+import "buf/alpha/image/v1/image.proto";
+
+enum ConvertFormat {
+  CONVERT_FORMAT_UNSPECIFIED = 0;
+  CONVERT_FORMAT_BIN = 1;
+  CONVERT_FORMAT_JSON = 2;
+}
+
+// ConvertService coverts serialized messages
+service ConvertService {
+  // Convert converts a serialized message according to
+  // the provided type name using either an image or a module_info.
+  rpc Convert(ConvertRequest) returns (ConvertResponse);
+}
+
+message ConvertRequest {
+  // type_name is the full type name of the serialized message (like acme.weather.v1.Units).
+  string type_name = 1;
+  // image is the image source that defines the serialized message.
+  //
+  // Cannot be provided if module_info is provided.
+  buf.alpha.image.v1.Image image = 2;
```

Comment:
It seems to me that `image` and `module_info` are either/or - do we really need to support both? It seems that we'd only support one, likely `image`, and leave building the `image` as an exercise for the caller. If we really need both, then we need to use `oneof`, but I don't like it

Reply:
we can leave building the image as an exercise for the caller, but this will add some overhead to the caller as they need to get the module, build it, and pass it in the request, instead of directly pass the module info and let us handle it in our side. And this might potentially makes using BSR module reference as source not a better option than other source (e.g. local image). How do you think?
We cannot use `oneof` in request field though, as it is disallowed by `go-apiclientgrpc` as Alex mentioned previously ([code](https://github.com/bufbuild/buf/blob/60cedc242b5231d92c37841278e223d6308a6637/private/pkg/protogenutil/protogenutil.go#L361))

Reply:
I'd prefer to keep it simple to start - a RPC endpoint should do one thing and do it well. Having an option between images and modules is a combination of build and convert, not just convert.

`go-apiclientgrpc` is something we wrote, we could always adjust that. But yea, keep just as `image` for now.

Reply:
sounds reasonable 👍

Reply:
I agree that a `oneof` is the right way to do this - we'll change `go-apiclientgrpc` later if we ever need to.

Requiring the client to compile the `Image` is fine for now, but we might eventually need to revisit this if we want other systems (not the `buf` CLI) to call this API. Otherwise they'd be forced to write their own `--type` -> `Image` conversion logic in whatever programming language their client is written is, which is impractical and error prone.

Regardless, we definitely need the `Image` parameter option so it makes sense to just start there and solve this later (if ever).

Reply:
I think if we want to revisit, we add a new endpoint, not add a `oneof` here.